### ISSUE TITLE
Analyzer for dependent static fields of value types.

### ISSUE THREAD

**Issue by 3ldar** [dotnet/roslyn#72164]
**Brief Description:**

In C#, static field initialization follows the source order. This means that incorrect ordering of dependent static fields can lead to null reference errors at runtime. However, when referenced fields are value types, the situation becomes more complex. Instead of null reference errors, you may encounter runtime bugs due to default values being assigned.

**Code Style Rule Description:**

Ensure correct initialization order for static fields, particularly when referenced fields are value types.

**Languages Applicable:**

Both C# and other applicable languages.

**Example Code that the Analyzer Should Report:**

```csharp
public static class Define
{
    // Warning: Incorrect initialization order for value types may lead to runtime bugs.
    public static readonly List<KeyValuePair<DefineChoices, string>> DefaultChoices = new([Choice1, Choice2]);
    public static readonly KeyValuePair<DefineChoices, string> Choice1 = new(DefineChoices.Choice1, "string1");
    public static readonly KeyValuePair<DefineChoices, string> Choice2 = new(DefineChoices.Choice2, "string2");
    public static readonly KeyValuePair<DefineChoices, string> Choice3 = new(DefineChoices.Choice1, "string3");
    public static readonly KeyValuePair<DefineChoices, string> Choice4 = new(DefineChoices.Choice2, "string4");

    public enum DefineChoices
    {
        Choice1,
        Choice2,
        Choice3,
        Choice4,
    }
}
```
The analyzer should warn about the above code piece as the referenced fields are value types and are initialized before being referenced. This could lead to runtime bugs, especially in scenarios where developers use the DefaultChoices collection keys for filtering another collection or in LINQ queries translated to SQL.
